lib/elixir/lib/dynamic_supervisor.ex

```diff
@@ -311,14 +311,26 @@ defmodule DynamicSupervisor do
   @doc since: "1.6.0"
   @spec start_child(Supervisor.supervisor(), Supervisor.child_spec() | {module, term} | module) ::
           on_start_child()
-  def start_child(supervisor, {_, _, _, _, _, _} = child_spec) do
-    validate_and_start_child(supervisor, child_spec)
+  @spec start_child(Supervisor.supervisor(), deprecated_child_spec :: Supervisor.Spec.spec()) ::
+          on_start_child()
+
+  def start_child(supervisor, %{id: _, start: _} = child_spec) do
+    validate_and_start_child(supervisor, Supervisor.child_spec(child_spec, []))
   end
 
-  def start_child(supervisor, child_spec) do
+  def start_child(supervisor, {module, _} = child_spec) when is_atom(module) do
     validate_and_start_child(supervisor, Supervisor.child_spec(child_spec, []))
   end
 
+  def start_child(supervisor, module) when is_atom(module) do
+    validate_and_start_child(supervisor, Supervisor.child_spec(module, []))
+  end
+
+  # Supports the deprecated Supervisor.Spec.spec/0 type
+  def start_child(supervisor, {_, _, _, _, _, _} = child_spec) do
```

Comment:
They are not really deprecated. We will continue to support them for legacy reasons, but they are discouraged.

Reply:
I will update the comment.